### PR TITLE
EMSUSD-1550 - Build with USD v24.08

### PR DESCRIPTION
**Pull request by seando-adsk**
#### EMSUSD-1550 - Build with USD v24.08
* Support for OneTBB (from USD build).

### CODE REVIEW

File: cmake/modules/FindMaya.cmake [hunk 1]
```diff
@@ -210,11 +210,6 @@ set(MAYA_LIBS_TO_FIND
     cgGL
     clew
 )
-if (CMAKE_BUILD_TYPE MATCHES Debug)
-    list(APPEND MAYA_LIBS_TO_FIND tbb_debug)
-else()
-    list(APPEND MAYA_LIBS_TO_FIND tbb)
-endif()
```
Comment (seando-adsk) [inline cmake/modules/FindMaya.cmake:217]: Even though we were finding TBB from Maya, it was never used from there. Just removed it (the library name changed with onetbb).

File: lib/mayaUsd/render/vp2RenderDelegate/meshViewportCompute.cpp
Comment (seando-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/meshViewportCompute.cpp:1]: Support OneTBB using the same pattern as what was done for Maya.

File: lib/mayaUsd/render/vp2RenderDelegate/renderDelegate.cpp [hunk 2]
```diff
@@ -38,7 +38,9 @@
 
 #include <maya/MProfiler.h>
 
+#if __has_include(<tbb/reader_writer_lock.h>)
 #include <tbb/reader_writer_lock.h>
+#endif
```
Comment (seando-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/renderDelegate.cpp:43]: This file doesn't exist in OneTBB.

File: lib/mayaUsd/render/vp2RenderDelegate/taskCommit.h [hunk 3]
```diff
@@ -60,7 +64,11 @@ template <typename Body> class HdVP2TaskCommitBody final : public HdVP2TaskCommi
     //! Release the memory using same allocator by calling destroy method.
     void destroy() override
     {
+#ifdef ONETBB_SPEC_VERSION
+        this->~HdVP2TaskCommitBody<Body>();
+#else
         my_allocator_type().destroy(this);
+#endif
```
Comment (seando-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/taskCommit.h:71]: Follow change done in Maya.

**Review by AramAzhari-adsk — APPROVED** [review]

**Review by AramAzhari-adsk — APPROVED** [review]

**Pull request merged**